packages/node-core/src/indexer/testing.service.ts

```diff
@@ -0,0 +1,265 @@
+    if (!(schemas as unknown as string[]).includes(schema)) {
+      await this.sequelize.createSchema(`"${schema}"`, undefined);
+    }
+
+    const modelRelations = getAllEntitiesRelations(this.project.schema);
+    await this.storeService.init(modelRelations, schema);
+    const tx = await this.sequelize.transaction();
+    this.storeService.setTransaction(tx);
+    const store = this.storeService.getStore();
+    sandbox.freeze(store, 'store');
+
+    // Init entities
+    logger.debug('Initializing entities');
+    await Promise.all(
+      test.dependentEntities.map((entity) => {
+        return entity.save();
+      })
+    );
+    await tx.commit();
+
+    logger.debug('Running handler');
+
+    try {
+      await this.indexBlock(block, test.handler);
+    } catch (e) {
+      this.totalFailedTests += test.expectedEntities.length;
+      logger.warn(`Test: ${test.name} field due to runtime error`, e);
+      this.failedTestsSummary.push({
+        testName: test.name,
+        entityId: undefined,
+        entityName: undefined,
+        failedAttributes: [`Runtime Error:\n${e.stack}`],
+      });
+      await this.sequelize.dropSchema(`"${schema}"`, {
+        logging: false,
+        benchmark: false,
+      });
+      return;
+    }
+
+    // Check expected entities
+    logger.debug('Checking expected entities');
+    let passedTests = 0;
+    let failedTests = 0;
+    for (let i = 0; i < test.expectedEntities.length; i++) {
+      const expectedEntity = test.expectedEntities[i];
+      const actualEntity = await store.get(expectedEntity._name, expectedEntity.id);
+      const attributes = actualEntity as unknown as CreationAttributes<Model>;
+      const failedAttributes: string[] = [];
+      let passed = true;
+      Object.keys(attributes).map((attr) => {
+        if (!this.isEqual(expectedEntity[attr], actualEntity[attr])) {
+          passed = false;
+          failedAttributes.push(
+            `\t\tattribute: "${attr}":\n\t\t\texpected: "${expectedEntity[attr]}"\n\t\t\tactual:   "${actualEntity[attr]}"`
+          );
+        }
+      });
+
+      if (passed) {
+        logger.info(`Entity check PASSED (Entity ID: ${expectedEntity.id})`);
+        passedTests++;
+      } else {
+        logger.warn(`Entity check FAILED (Entity ID: ${expectedEntity.id})`);
+        this.failedTestsSummary.push({
+          testName: test.name,
+          entityId: expectedEntity.id,
+          entityName: expectedEntity._name,
+          failedAttributes: failedAttributes,
+        });
+
+        failedTests++;
+      }
+    }
+
+    this.totalPassedTests += passedTests;
+    this.totalFailedTests += failedTests;
+
+    logger.info(
+      `Test: ${test.name} completed with ${chalk.green(`${passedTests} passed`)} and ${chalk.red(
+        `${failedTests} failed`
+      )} checks`
+    );
+    await this.sequelize.dropSchema(`"${schema}"`, {
+      logging: false,
+      benchmark: false,
+    });
```

Comment:
I think we could move this into a `finally` so we can guarantee that it always happens. If there's any error checking the entities currently then this won't get called